client/src/components/composite/Admin/AdminEventView/AdminAllEvents/AdminAllEvents.tsx

```diff
@@ -0,0 +1,260 @@
+        const { physical_start_date, physical_end_date } = event
+        if (
+          EventDateComparisons.isPastEvent(
+            new Date(DateUtils.timestampMilliseconds(physical_start_date)),
+            physical_end_date &&
+              new Date(DateUtils.timestampMilliseconds(physical_end_date))
+          )
+        ) {
+          buf.pastEvents.push(event)
+        } else {
+          buf.upcomingAndCurrentEvents.push(event)
+        }
+
+        /**
+         * Start dates ascending for upcoming and current events
+         */
+        buf.upcomingAndCurrentEvents.sort(
+          (
+            { physical_start_date: startDate1 },
+            { physical_start_date: startDate2 }
+          ) =>
+            DateUtils.timestampMilliseconds(startDate1) -
+            DateUtils.timestampMilliseconds(startDate2)
+        )
+
+        return buf
+      },
+      { upcomingAndCurrentEvents: [], pastEvents: [] }
+    )
+  }, [rawEvents])
+
+  const selectedEventObject = useMemo(() => {
+    if (!selectedEventId) return
+
+    const eventInfo = rawEvents.find((event) => event.id === selectedEventId)
+
+    /**
+     * See if we can find the event, otherwise give up
+     * Has a side-effect depending on {@link rawEvents}
+     */
+    if (!eventInfo) {
+      fetchMoreEvents?.()
+      // We can't fetch any more, and it is still undefined, so we remove the query
+      !isLoading && !hasMoreEvents && eventSelectionHandler(undefined)
+    }
+
+    return eventInfo
+  }, [
+    selectedEventId,
+    rawEvents,
+    fetchMoreEvents,
+    hasMoreEvents,
+    isLoading,
+    eventSelectionHandler
+  ])
+
+  /**
+   * Detailed view of the event
+   */
+  const SelectedEventPanel = useMemo(() => {
+    if (!selectedEventObject) {
+      return <Loader />
+    }
+
+    const {
+      sign_up_start_date,
+      google_forms_link,
+      physical_end_date,
+      physical_start_date,
+      description,
+      title
+    } = selectedEventObject
+    return (
+      <EventDetailed
+        onBack={() => {
+          eventSelectionHandler(undefined)
+        }}
+        date={EventMessages.eventDateRange(
+          new Date(DateUtils.timestampMilliseconds(physical_start_date)),
+          physical_end_date &&
+            new Date(DateUtils.timestampMilliseconds(physical_end_date))
+        )}
+        isPastEvent={EventDateComparisons.isPastEvent(
+          new Date(DateUtils.timestampMilliseconds(physical_start_date)),
+          physical_end_date &&
+            new Date(DateUtils.timestampMilliseconds(physical_end_date))
+        )}
+        image={selectedEventObject.image_url || IMAGE_PLACEHOLDER_SRC}
+        location={selectedEventObject.location}
+        signUpOpenDate={
+          new Date(DateUtils.timestampMilliseconds(sign_up_start_date))
+        }
+        googleFormLink={google_forms_link}
+        content={<p>{description}</p>}
+        title={title}
+      />
+    )
+  }, [selectedEventObject, eventSelectionHandler])
```

Comment:
we don't need this

Reply:
I replaced the use with `null` on line 153